internal/storage/download_test.go

```diff
@@ -56,7 +63,47 @@ func TestDownloadToFile(t *testing.T) {
 			HashType:  "SHA256",
 			HashValue: "4eefb9a7a40a8b314b586a00f307157043c0bbe4f59fa39cba88773680758bc3",
 		},
+	}, t)
+}
+
+// TestDownloadToFileSecure tests downloadToFile function, using secure protocol(s).
+func TestDownloadToFileSecure(t *testing.T) {
+	if runtime.GOOS != "linux" {
+		t.Skip("this test only runs on Linux.")
 	}
+	sslCertFile := os.Getenv("SSL_CERT_FILE")
+	certFiles := strings.Split(sslCertFile, string(os.PathListSeparator))
+	containsTestCert := false
+	for _, file := range certFiles {
+		if strings.HasSuffix(strings.TrimSpace(file), filepath.Base(testCertFile)) {
+			containsTestCert = true
+			break
+		}
+	}
+	if !containsTestCert {
```

Comment:
What are we validating with this check?

Reply:
The test relies upon system certificate pool. Adding test certificate to it, depends on parsing an environment variable, which may not work on some Linux distributions. So it is better to check if certificate has been added.